_reports/2021-05-88mph.md

```diff
@@ -0,0 +1,285 @@
+
+```
+function withdrawAndDoSomething() {
+    contract.withdrawVested(address(this), amount);
+    token.transfer(externalWallet, amount)
+}
+```
+
+If the contract has no other functions to transfer out funds, they may be locked forever in this contract.
+
+Recommend not allowing users to withdraw on behalf of other users.
+
+**[ZeframLou (88mph) acknowledged](https://github.com/code-423n4/2021-05-88mph-findings/issues/15#issuecomment-844447528):**
+> This is true, but `Vesting.sol` is only kept for legacy support, `Vesting02.sol` will be the main vesting contract, so we're fine with this.
+
+## [[L-03] Extra precautions in updateAndQuery](https://github.com/code-423n4/2021-05-88mph-findings/issues/10)
+
+The function `updateAndQuery` of EMAOracle.sol subtracts the `incomeIndex` with the previous `incomeIndex`.
+These `incomeIndex` values are retrieved via the moneyMarket contract from an external contract.
+
+If by accident the previous `incomeIndex` is larger than the current `incomeIndex` then the subtraction would be negative and the code halts (reverts), without an error message.
+
+Also the `updateAndQuery` function would not be able to execute (until the current `incomeIndex` is larger than the previous `incomeIndex`).
+
+This situation could occur when an error occurs in one of the current or future money markets.
+
+Recommend giving an error message when the previous `incomeIndex` is larger than the current `incomeIndex` and/or create a way to recover from this erroneous situation.
+
+**[ZeframLou (88mph) confirmed](https://github.com/code-423n4/2021-05-88mph-findings/issues/10#issuecomment-844618871):**
+> Fixed in https://github.com/88mphapp/88mph-contracts/commit/9de33956c1c31cde52768110dc6ec9c19049a748
+
+## [[L-04] Add extra error message in_depositRecordData](https://github.com/code-423n4/2021-05-88mph-findings/issues/7)
+
+In the function `_depositRecordData` of DInterest.sol, `interestAmount` is lowered with `feeAmount`.
+
+If by accident `feeAmount` happens to be larger than `interestAmount` an error occurs and the execution stops, without an error message.
+
+This might make troubleshooting this situation more difficult.
+
+Suggest adding something like:
+```
+require(interestAmount >= feeAmount,"DInterest: fee too large");
+```
+
+**[ZeframLou (88mph) acknowledged](https://github.com/code-423n4/2021-05-88mph-findings/issues/7)**
+
+## [[L-05] function payInterestToFunders does not have a re-entrancy modifier](https://github.com/code-423n4/2021-05-88mph-findings/issues/6)
+
+Function `payInterestToFunders` does not have a re-entrancy modifier. I expect to see this modifier because similar functions (including sponsored version) have it.
+
+Add 'nonReentrant' to function `payInterestToFunders`.
+
+**[ZeframLou (88mph) confirmed](https://github.com/code-423n4/2021-05-88mph-findings/issues/6#issuecomment-844461166):**
+> Fixed in https://github.com/88mphapp/88mph-contracts/commit/ed9dec252b1dde4225f4bfc1ac4a52bc79af40a4
```

Comment:
All links to PRs, etc, should be linkified rather than plain text urls. eg "Fixed in [this PR](link-here)" or "Fixed in [this commit](link-here)"